api/v1beta2/awsmachine_webhook.go

```diff
@@ -361,6 +362,14 @@ func (r *AWSMachine) validateNetworkElasticIPPool() field.ErrorList {
 	return allErrs
 }
 
+func (r *AWSMachine) validateInstanceMarketType() field.ErrorList {
+	var allErrs field.ErrorList
+	if ptr.Deref(r.Spec.MarketType, "") == MarketTypeCapacityBlock && r.Spec.SpotMarketOptions != nil {
```

Comment:
shouldn't this also check MarketType is not ondemand if SpotMarketOptions != nil? i.e. the only valid when SpotMarketOptions != nil is MarketType=spot

Reply:
oh yeah good catch thanks